src/cryptonote_core/tx_pool.cpp

```diff
@@ -1388,7 +1428,7 @@ namespace cryptonote
           remove.push_back(txid);
           return true;
         }
-        if (!insert_key_images(tx, txid, meta.kept_by_block))
+        if (!insert_key_images(tx, txid, meta.get_relay_method()))
         {
           MFATAL("Failed to insert key images from txpool tx");
           return false;
```

Comment:
So `anonymity` != `stem`, which might be part of the confusion. The next PR in progress now, adds `stem` and `fluff`.

`anonymity` means the tx originated at the node, and that node had i2p/tor enabled. It has to be different than `stem`, because txes received over clearnet `stem` need to be relayed the same way (or fluffed).

Reply:
You are/were correct, but I'm not sure why the comment was placed here. Perhaps because this check passes and then the tx is added to `m_txs_by_fee_and_receive_time`? This was another bug :/

My current approach is to add to `m_txs_by_fee_and_receive_time` for consistency, and then filter out on in the block_template function since the metadata is already being retrieved.  The primary purpose of this member variable is for block templates, so it could probably be omitted entirely (although that would cause spurious messages about it missing to come up).